src/parsers/message.rs

```diff
@@ -282,6 +284,12 @@ impl MessageParser {
                                 || content_type
                                     .map_or_else(|| true, |c| !c.has_attribute("name")))));
 
+                // if message consists of single text/plain part, classify as text regardless
+                // of encoding issues: see malformed/018.eml
+                let is_inline =
+                    state.parts == 1 && mime_type == MimeType::TextPlain && is_encoding_problem
+                        || is_inline;
+
```

Comment:
I would do this instead:

``is_inline || is_encoding_problem && state.parts == 1 && mime_type == MimeType::TextPlain``